core/chains/evm/gas/suggested_price_estimator.go

```diff
@@ -19,68 +18,69 @@ import (
 )
 
 var (
-	_ EvmEstimator = &l2SuggestedPriceEstimator{}
+	_ EvmEstimator = &SuggestedPriceEstimator{}
 )
 
 //go:generate mockery --quiet --name rpcClient --output ./mocks/ --case=underscore --structname RPCClient
 type rpcClient interface {
 	CallContext(ctx context.Context, result interface{}, method string, args ...interface{}) error
 }
 
-// l2SuggestedPriceEstimator is an Estimator which uses the L2 suggested gas price from eth_gasPrice.
-type l2SuggestedPriceEstimator struct {
-	services.StateMachine
+// SuggestedPriceEstimator is an Estimator which uses the L2 suggested gas price from eth_gasPrice.
+type SuggestedPriceEstimator struct {
+	utils.StartStopOnce
 
 	client     rpcClient
 	pollPeriod time.Duration
 	logger     logger.Logger
 
 	gasPriceMu sync.RWMutex
-	l2GasPrice *assets.Wei
+	GasPrice   *assets.Wei
 
 	chForceRefetch chan (chan struct{})
 	chInitialised  chan struct{}
 	chStop         utils.StopChan
 	chDone         chan struct{}
 }
 
-// NewL2SuggestedPriceEstimator returns a new Estimator which uses the L2 suggested gas price.
-func NewL2SuggestedPriceEstimator(lggr logger.Logger, client rpcClient) EvmEstimator {
-	return &l2SuggestedPriceEstimator{
+// NewSuggestedPriceEstimator returns a new Estimator which uses the L2 suggested gas price.
+// Can also be used for non-L2 chains.
```

Comment:
We can do the same here and I think that's generic enough that we don't need the second comment
```suggestion
// NewSuggestedPriceEstimator returns a new Estimator which uses the suggested gas price.
```